src/containers/auth/forgotPassword/index.jsx

```diff
@@ -0,0 +1,168 @@
+import React, { PureComponent } from 'react';
+import Modal from 'react-modal';
+import { FaTimes } from 'react-icons/fa';
+import { IoIosCheckmarkCircle } from 'react-icons/io';
+
+import PropTypes from 'prop-types';
+
+import styles from './styles.module.scss';
+import { forgotPasswordRequest } from '../../../services/auth';
+import validateAuth from '../../../validation/auth';
+import ShowMessage from '../showMessage';
+import Spinner from '../../../components/spinner';
+
+Modal.setAppElement('#root');
+
+class ForgotPassword extends PureComponent {
+  state = {
+    status: null,
+    data: null,
+    email: null,
+    message: null,
+    pending: false,
+  };
+
+  send = async email => {
+    const errors = validateAuth({ email });
+    this.setState({
+      pending: true,
+    });
+    if (!errors.email) {
+      const response = await forgotPasswordRequest({ email });
+      const { data, status } = response;
+      this.setState({
+        status,
+        data,
+      });
+    } else {
+      this.setState({
+        message: errors.email,
+      });
+    }
+    this.setState({
+      pending: false,
+    });
+  };
+
+  onChangeInput = e => {
+    this.setState({
+      email: e.target.value,
+    });
+  };
+
+  onCloseModalSetDefault = () => {
+    this.setState({
+      status: null,
+      data: null,
+      email: null,
+      message: null,
+      pending: false,
+    });
+  };
+
+  onSubmit = e => {
+    e.preventDefault();
+    this.send(this.state.email);
+  };
+
+  onInput = () =>
+    this.setState({
+      message: null,
+    });
+
+  modalClose = () => {
+    this.onCloseModalSetDefault();
+    this.props.onModalClose(false);
+  };
+
+  onRequestClose = () => {
+    this.props.onModalClose(false);
+  };
+
+  render() {
+    const {
+      input,
+      btn,
+      modal,
+      form,
+      row,
+      row_flex_end: rowFlexEnd,
+      row_top: rowTop,
+      close_modal: closeModal,
+      success_message: successMessage,
+      error_message: errorMessage,
+      big_size_message: bigSizeMessage,
+    } = styles;
+
+    const { status, data, message, pending } = this.state;
+
+    const { modalStatus } = this.props;
+
+    let showValidationMessage = !!message;
+    let showForm = !status && !pending;
+    let showSpinner = pending;
+    let showErrorMessage = (status >= 300);
+    let showSuccessMessage = (status < 300);
+    let showCheckMark = !status;
```

Comment:
I don't like how you implement logic here. A lot of variables, all logic connected with statuses, that kind a strange. Maybe you should try to do it better. 
Also use const instead of let